sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformation.scala

```diff
@@ -67,6 +67,14 @@ case class ScriptTransformation(
       val cmd = List("/bin/bash", "-c", script)
       val builder = new ProcessBuilder(cmd.asJava)
 
+      // for compatibility with hive
+      // if the script is a python/bash/... file, we should make it executable
+      if (new File(script).exists) {
+        new File(script).setExecutable(true)
```

Comment:
Doesn't this create a huge security hole? Especially because we are not resetting this to its old state.